omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java

```diff
@@ -0,0 +1,225 @@
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+import javax.servlet.ServletContext;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@Resource(name = RestConstants.VERSION_1 + "/moduleaction", supportedClass = ModuleAction.class, supportedOpenmrsVersions = {
+        "1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*" })
+public class ModuleActionResource1_8 extends BaseDelegatingResource<ModuleAction> implements Creatable {
+	
+	/**
+	 * ModuleFactoryWrapper is used for testing purposes.
+	 */
+	private ModuleFactoryWrapper moduleFactoryWrapper = new ModuleFactoryWrapper();
+	
+	public void setModuleFactoryWrapper(ModuleFactoryWrapper moduleFactoryWrapper) {
+		this.moduleFactoryWrapper = moduleFactoryWrapper;
+	}
+	
+	/**
+	 * Overriding create directly, because ModuleFactory requires ServletContext to execute any
+	 * action
+	 */
+	@Override
+	public Object create(SimpleObject post, RequestContext context) throws ResponseException {
+		moduleFactoryWrapper.checkPrivilege();
+		
+		ModuleAction action = newDelegate();
+		setConvertedProperties(action, post, getCreatableProperties(), true);
+		
+		Collection<Module> modules;
+		if (action.isAllModules() != null && action.isAllModules()) {
+			modules = moduleFactoryWrapper.getLoadedModules();
+			action.setModules(new ArrayList<Module>(modules));
+		} else {
+			modules = action.getModules();
+		}
+		
+		ServletContext servletContext = getServletContext(context);
+		
+		if (modules == null || modules.isEmpty()) {
+			throw new IllegalRequestException("Cannot execute action " + action.getAction() + " on empty set of modules.");
+		} else {
+			//rest module cannot be stopped/unloaded during handling request
+			Module restModule = moduleFactoryWrapper.getModuleById(RestConstants.MODULE_ID);
+			modules.remove(restModule);
+			switch (action.getAction()) {
+				case START:
+					startModules(modules, servletContext);
+					break;
+				case STOP:
+					stopModules(modules, servletContext, true);
+					break;
+				case RESTART:
+					restartModules(modules, servletContext);
+					break;
+				case UNLOAD:
+					unloadModules(modules, servletContext);
+					break;
+			}
+		}
+		
+		return ConversionUtil.convertToRepresentation(action, Representation.DEFAULT);
+	}
+	
+	private void restartModules(Collection<Module> modules, ServletContext servletContext) {
+		stopModules(modules, servletContext, false);
+		startModules(modules, servletContext);
+	}
+	
+	private void unloadModules(Collection<Module> modules, ServletContext servletContext) {
+		boolean needsRefresh = false;
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStarted(module)) {
+				moduleFactoryWrapper.stopModuleSkipRefresh(module, servletContext);
+				needsRefresh = true;
```

Comment:
I see, in the legacy ui we didnt have this start multiple feature, so that makes sense then